src/main/java/com/microfocus/application/automation/tools/mc/JobConfigurationProxy.java

```diff
@@ -51,75 +53,104 @@
     private JobConfigurationProxy() {
     }
 
-    public static JobConfigurationProxy getInstance(){
-        if(instance == null){
+    public static JobConfigurationProxy getInstance() {
+        if (instance == null) {
             instance = new JobConfigurationProxy();
         }
         return instance;
     }
     //Login to MC
-    public JSONObject loginToMC(String mcUrl, String mcUserName, String mcPassword, String mcTenantId,
-                                String proxyAddress, String proxyUsername, String proxyPassword) {
+    public JSONObject loginToMC(String mcUrl, AuthModel authModel, ProxySettings proxy) {
 
         JSONObject returnObject = new JSONObject();
         try {
             Map<String, String> headers = new HashMap<String, String>();
             headers.put(Constants.ACCEPT, "application/json");
             headers.put(Constants.CONTENT_TYPE, "application/json;charset=UTF-8");
-//            headers.put("TENANT_ID_COOKIE", mcTenantId);
 
             JSONObject sendObject = new JSONObject();
-            if(!StringUtils.isNullOrEmpty(mcTenantId)){
-                mcUserName = mcUserName + "#" + mcTenantId;
+            if (null == proxy) {
+                proxy = new ProxySettings();
             }
-            sendObject.put("name", mcUserName);
-            sendObject.put("password", mcPassword);
-            sendObject.put("accountName", "default");
-            HttpResponse response = HttpUtils.post(HttpUtils.setProxyCfg(proxyAddress, proxyUsername, proxyPassword), mcUrl + Constants.LOGIN_URL, headers, sendObject.toJSONString().getBytes());
-
-            if (response != null && response.getHeaders() != null) {
-                Map<String, List<String>> headerFields = response.getHeaders();
-                List<String> hp4mSecretList = headerFields.get(Constants.LOGIN_SECRET);
-                String hp4mSecret = null;
-                if (hp4mSecretList != null && hp4mSecretList.size() != 0) {
-                    hp4mSecret = hp4mSecretList.get(0);
+            HttpResponse response;
+            if ("base".equals(authModel.getValue())) {
+                if (!StringUtils.isNullOrEmpty(authModel.getMcTenantId())) {
+                    authModel.setMcUserName(authModel.getMcUserName() + "#" + authModel.getMcTenantId());
```

Comment:
This will cause the original username to change, works for login, but will cause failure in the test run. Please use temporary variables, do not change the original input object's attributes.